lib/train/transports/cisco_ios.rb

```diff
@@ -0,0 +1,147 @@
+      validate_options(@options)
+      @connection ||= Connection.new(@options)
+    end
+
+    class Connection < BaseConnection
+      def initialize(options)
+        super(options)
+
+        @host = @options.delete(:host)
+        @user = @options.delete(:user)
+        @port = @options.delete(:port)
+
+        @enable_password = @options.delete(:enable_password)
+
+        @prompt = /^\S+[>#]\r\n.*$/
+      end
+
+      def uri
+        "ssh://#{@user}@#{@host}:#{@port}"
+      end
+
+      private
+
+      def establish_connection
+        logger.debug("[SSH] opening connection to #{self}")
+
+        @ssh = Net::SSH.start(
+          @host,
+          @user,
+          @options.delete_if { |_key, value| value.nil? },
+        )
+
+        @channel ||= open_channel
+
+        # Escalate privilege to enable mode if password is given
+        if @enable_password
+          run_command_via_channel("enable\r\n#{@enable_password}")
+        end
+
+        # Prevent `--MORE--` by removing terminal length limit
+        run_command_via_channel('terminal length 0')
+
+        @ssh
+      end
+
+      def run_command_via_connection(cmd)
+        @session ||= establish_connection
+
+        result = run_command_via_channel(cmd)
+        CommandResult.new(*format_result(result))
+      end
+
+      def format_result(result)
+        stderr_with_exit_1 = ['', result, 1]
+        stdout_with_exit_0 = [result, '', 0]
+
+        # IOS commands do not have an exit code, so we must capture known errors
+        case result
+        when /Bad IP address/
+          stderr_with_exit_1
+        when /Incomplete command/
+          stderr_with_exit_1
+        when /Invalid input detected/
+          stderr_with_exit_1
+        when /Unrecognized host/
+          stderr_with_exit_1
+        else
+          stdout_with_exit_0
+        end
+      end
+
+      def run_command_via_channel(cmd)
+        # Ensure buffer is empty before sending data
+        @buf = ''
+
+        logger.debug("[SSH] Running `#{cmd}` on #{self}")
+        @channel.send_data(cmd + "\r\n")
+
+        logger.debug('[SSH] waiting for prompt')
+        until @buf =~ @prompt
+          raise BadEnablePassword if @buf =~ /Bad secrets/
+          @channel.connection.process(0)
+        end
+
+        # Save the buffer and clear it for the next command
+        output = @buf.dup
+        @buf = ''
+
+        format_output(output, cmd)
+      end
+
+      # The buffer (@buf) contains all data sent/received on the SSH channel so
+      # we need to format the data to match what we would expect from Train
+      def format_output(output, cmd)
+        # Remove leading prompt
+        output.sub!(/(\r\n|^)\S+[>#]/, '')
+
+        # Remove command string
+        output.sub!(/#{cmd}\r\n/, '')
+
+        # Remove trailing prompt
+        output.gsub!(/\S+[>#](\r\n|$)/, '')
+
+        # Remove trailing returns/newlines
+        output.gsub!(/(\r\n)+$/, '')
+
+        output
+      end
```

Comment:
`String#sub` and `String#gsub` return the modified string, so you can chain them. Also, giving the regular expression matchers explaining variables can help avoid an almost certain fate of the regexp being changed, but not the comment that explains it.
```ruby
def format_output(output, cmd)
  leading_prompt        = /(\r\n|^)\S+[>#]/
  command_string        = /#{cmd}\r\n/
  trailing_prompt       = /\S+[>#](\r\n|$)/
  trailing_line_endings = /(\r\n)+$/

  output
    .sub(leading_prompt, '')
    .sub(command_string, '')
    .gsub(trailing_prompt, '')
    .gsub(trailing_line_endings, '')
end
```